ostree-upload-server.py

```diff
@@ -251,16 +276,18 @@ def webapp_callback(task_name, filepath):
                         logging.debug("idle, do maintenance")
                         workers.stop()
 
-                        try:
-                            output = check_output(["flatpak",
-                                                   "build-update-repo",
-                                                   "--generate-static-deltas",
-                                                   "--prune",
-                                                   self._repo],
-                                                  stderr=STDOUT)
-                            logging.info("completed maintenance: " + output)
-                        except CalledProcessError as e:
-                            logging.error("failed maintenance: " + e.output)
+                        with RepoLock(self._repo, exclusive=True):
+                            try:
+                                output = check_output(["flatpak",
+                                                       "build-update-repo",
+                                                       "--generate-static-deltas",
+                                                       "--prune",
```

Comment:
I don't know if we want to be generating static deltas or pruning randomly. Those are both very costly procedures. Are new tasks blocked during this time?

Also, is the summary regenerated after each import? It would kind of suck to have the client complete an upload but not have it appear to be available in the repo.